lib/polymorphic_integer_type/activerecord_5_0_0/polymorphic_array_value_extension.rb

```diff
@@ -1,17 +1,31 @@
 module PolymorphicIntegerType
   module PolymorphicArrayValueExtension
+
+    # original method:
+    # def type_to_ids_mapping
+    #   default_hash = Hash.new { |hsh, key| hsh[key] = [] }
+    #   result = values.each_with_object(default_hash) do |value, hash|
+    #     hash[klass(value).polymorphic_name] << convert_to_id(value)
+    #   end
+    # end
+
     def type_to_ids_mapping
-      super.tap do |result|
-        association = @associated_table.send(:association)
-        klass = association.active_record
-        name = association.name
+      association = @associated_table.send(:association)
+      name = association.name
+      default_hash = Hash.new { |hsh, key| hsh[key] = [] }
+      values.each_with_object(default_hash) do |value, hash|
+        klass = value.class
```

Comment:
Should this be `klass(value)` (from looking at the original code?)

Reply:
I originally had that, but turns out this magical `#klass` method doesn't exist prior to 5.2 so it blows up under 5.1. The end result is the same though so this was a fix to make it work under both 🤷‍♂️